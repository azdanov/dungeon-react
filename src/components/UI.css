.ui {
  position: absolute;
  right: 0;
  top: 20%;
  display: flex;
  justify-content: space-between;
  flex-direction: column;
  width: 420px
}

.box {
  background-color: rgba(224, 224, 224, 1);
  padding: 10px;
  margin-top: 20px;
}

.stats {
  display: flex;
  justify-content: space-between;
  align-items: baseline;
}

.stats progress {
  display: block;
  width: 140px;
}

.stats ul:nth-of-type(2) {
  margin-right: 15px;
}

.weapon {
  margin-left: 24px;
}

.box li {
  margin-top: 5px;
  padding: 2px;
}

.log {
}