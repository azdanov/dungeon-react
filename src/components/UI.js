// @flow

import React from 'react';
import './UI.css';

function pickIcon(id: number) {
  switch (id) {
    case 0:
      return (
        <li>
          Knife {' '}
          <span role="img" aria-label="knife">
            🔪
          </span>
        </li>
      );
    case 1:
      return (
        <li>
          Bomb {' '}
          <span role="img" aria-label="bomb">
            💣
          </span>
        </li>
      );
    case 2:
      return (
        <li>
          Telephone {' '}
          <span role="img" aria-label="telephone">
            📞
          </span>
        </li>
      );
    case 3:
      return (
        <li>
          Flashlight {' '}
          <span role="img" aria-label="flashlight">
            🔦
          </span>
        </li>
      );
    case 4:
      return (
        <li>
          Newspaper {' '}
          <span role="img" aria-label="newspaper">
            🗞️
          </span>
        </li>
      );
    case 5:
      return (
        <li>
          Pen {' '}
          <span role="img" aria-label="pen">
            🖊️
          </span>
        </li>
      );
    case 6:
      return (
        <li>
          Scissors {' '}
          <span role="img" aria-label="scissors">
            ✂️
          </span>
        </li>
      );
    case 7:
      return (
        <li>
          Pick {' '}
          <span role="img" aria-label="pick">
            ⛏️
          </span>
        </li>
      );
    case 8:
      return (
        <li>
          Dagger {' '}
          <span role="img" aria-label="dagger">
            🗡️
          </span>
        </li>
      );
    case 9:
      return (
        <li>
          Wrench {' '}
          <span role="img" aria-label="wrench">
            🔧
          </span>
        </li>
      );
    case 10:
      return (
        <li>
          Hammer {' '}
          <span role="img" aria-label="hammer">
            🔨
          </span>
        </li>
      );
    default:
      return '';
  }
}

const UI = (props: {
  log: Array<string>,
  health: number,
  strength: number,
  level: number,
  experience: number,
  levelThreshold: number,
  zone: number,
  weapon: { id: number, power: number },
}) => (
  <div className="ui">
    <div className="box stats">
      <ul>
        <li>
          <span role="img" aria-label="red heart">
            ❤️
          </span>{' '}
          Health: {props.health}
        </li>
        <li>
          <span role="img" aria-label="curled biceps">
            💪
          </span>{' '}
          Strength: {props.strength}
        </li>
        <li>
          <span role="img" aria-label="level arrow">
            🔺
          </span>{' '}
          Level: {props.level}
          <progress value={props.experience} max={props.levelThreshold}>
            {props.experience} %
          </progress>
        </li>
        <li>
          <span role="img" aria-label="zone symbol">
            🔹
          </span>{' '}
          Zone: {props.zone}
        </li>
      </ul>
      <ul>
        <li className="inventory">
          <span role="img" aria-label="backpack">
            🎒
          </span>{' '}
          Inventory:{' '}
          <ul className="weapon">
            {pickIcon(props.weapon.id)}
            <li>Damage: {props.weapon.power}</li>
          </ul>
        </li>
      </ul>
    </div>
    <div className="box">
      <ul className="log">
        <li>{props.log[props.log.length - 1]}</li>
        <li>{props.log[props.log.length - 2]}</li>
        <li>{props.log[props.log.length - 3]}</li>
        <li>{props.log[props.log.length - 4]}</li>
      </ul>
    </div>
  </div>
);

export default UI;
