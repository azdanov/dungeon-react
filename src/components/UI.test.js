// @flow
import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { createSerializer } from 'enzyme-to-json';

import UI from './UI';

configure({ adapter: new Adapter() });

expect.addSnapshotSerializer(createSerializer({ mode: 'deep' }));

describe('UI', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <UI
        log={['Message 1', 'Message 2', 'Message 3', 'Message 4']}
        health={100}
        strength={10}
        level={1}
        experience={50}
        levelThreshold={120}
        zone={1}
        weapon={{ id: 1, power: 10 }}
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
