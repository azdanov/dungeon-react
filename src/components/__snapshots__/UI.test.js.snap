// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI renders correctly 1`] = `
<div
  className="ui"
>
  <div
    className="box stats"
  >
    <ul>
      <li>
        <span
          aria-label="red heart"
          role="img"
        >
          ❤️
        </span>
         
        Health: 
        100
      </li>
      <li>
        <span
          aria-label="curled biceps"
          role="img"
        >
          💪
        </span>
         
        Strength: 
        10
      </li>
      <li>
        <span
          aria-label="level arrow"
          role="img"
        >
          🔺
        </span>
         
        Level: 
        1
        <progress
          max={120}
          value={50}
        >
          50
           %
        </progress>
      </li>
      <li>
        <span
          aria-label="zone symbol"
          role="img"
        >
          🔹
        </span>
         
        Zone: 
        1
      </li>
    </ul>
    <ul>
      <li
        className="inventory"
      >
        <span
          aria-label="backpack"
          role="img"
        >
          🎒
        </span>
         
        Inventory:
         
        <ul
          className="weapon"
        >
          <li>
            Bomb 
             
            <span
              aria-label="bomb"
              role="img"
            >
              💣
            </span>
          </li>
          <li>
            Damage 
            10
          </li>
        </ul>
      </li>
    </ul>
  </div>
  <div
    className="box"
  >
    <ul
      className="log"
    >
      <li>
        Message 2
      </li>
      <li>
        Message 1
      </li>
      <li>
        Message 4
      </li>
      <li>
        Message 3
      </li>
    </ul>
  </div>
</div>
`;
